x/staking/keeper/msg_server.go

```diff
@@ -99,6 +81,32 @@ func (k msgServer) CreateValidator(ctx context.Context, msg *types.MsgCreateVali
 				"got: %s, expected: %s", pk.Type(), cp.Validator.PubKeyTypes,
 			)
 		}
+
+		if pkType == "ed25519" && len(pk.Bytes()) != ed25519.PubKeySize {
```

Comment:
nit, can you use the constant for the PK type?

Reply:
I have the same idea as you, but I can't find the corresponding constant definition.

Reply:
done